src/geanyobject.c

```diff
@@ -156,6 +156,12 @@ static void create_signals(GObjectClass *g_object_class)
 		G_SIGNAL_RUN_FIRST,
 		0, NULL, NULL, g_cclosure_marshal_VOID__VOID,
 		G_TYPE_NONE, 0);
+	geany_object_signals[GDB_PROJECT_BEFORE_CLOSE] = g_signal_new (
+		"project-before-close",
+		G_OBJECT_CLASS_TYPE (g_object_class),
+		G_SIGNAL_RUN_FIRST,
+		0, NULL, NULL, g_cclosure_marshal_VOID__VOID,
+		G_TYPE_NONE, 0);
```

Comment:
That would be fine with me. I mostly just didn't want to mess with the existing signal in case some plugins somehow depended on the existing behaviour.